### DISCUSSION TITLE
[Bug] SpringCloudStream bind rabbitMQ the traceId is lost

### DISCUSSION THREAD

**Discussion by liuxiansheng990710** [apache/skywalking#11705] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

- SkyWalking APM Version：8.5.0
- Problem description:When using springCloudStream to bind rabbitMQ, the new version consumer is the function method, and the traceID is lost
- This trace is already present in skywalking ui
  ![image](https://github.com/apache/skywalking/assets/77720013/b7e0ff7d-a7e7-46b8-9c25-0b4ee721a863)
> producerLog
  ```java
   INFO 2023-12-28 18:03:08,868 13716 - [TID: e6a749c792bf41dbb8bae517c4079b74.173.17037577877320001] - <mq_procuder> {"origin":"provider-auth","environment":"alpha"
 ```

> consumerLog
  ```java
   INFO 2023-12-28 18:03:09,120 10904 - [TID: N/A] - [test-topic.provider-base-1]  - <mq_consumer> - {"origin":"provider-base","environment":"alpha"
 ```
> springCloudStream - RabbitMQ -Cousumer class
  ```java
    @Bean("21-alpha-test")
    public Consumer<Message<MQTestModel>> mqInputTest() 
```

> supplement

Consumer service other logs printed normally, only lost traceId when serving as rabbitmq consumer

### What you expected to happen

When consumers do not lose traceId

### How to reproduce

Use springcloudstream for rabbitmq binding, then send a message using the latest version of the function type to return the value as the consumer, in which log.info()

Assuming that the log4j2 plugin has been used to configure [% traceId]

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Log only has traced ID after the span of consuming span created. 
Logs are never always having trace ID. Don't expect that.

**Reply by liuxiansheng990710** [reply]
Is there any way to manually pass traceId

**Reply by wu-sheng** [reply]
Your trace is having both sides, why manual? It doesn't make sense.

**Reply by liuxiansheng990710** [reply]
Although there are two sides in the UI, the tid is missing in the consumerLog. I still want to print it out

**Reply by wu-sheng** [reply]
That means, your logs are not in the same thread of the consuming. And I don't know why. 
As mentioned many times, maintainers don't have all knowledge about plugins. You could ask to RocketMQ community, someone from there writes the plugin, we just host the codes for them.

**Reply by liuxiansheng990710** [reply]
I thought of cross threading, so I used @ TraceCrossThread, but it was still lost,
I even imitated RunnableWrapper. of (), but there was no traciId present，
Is there currently no solution to this problem, or is there a manual cross thread delivery method

**Reply by wu-sheng** [reply]
Again, ask RocketMQ community for the help.
